packages/contract/lib/execute.js

```diff
@@ -22,20 +23,34 @@ const execute = {
     const interfaceAdapter = this.interfaceAdapter;
 
     return new Promise(function (accept) {
-      // Always prefer specified gas - this includes gas set by class_defaults
+      // Always prefer gas specified by user (if a user sets gas to 0, that is treated
+      // as undefined here and we do proceed to do gas estimation)
       if (params.gas) return accept(params.gas);
       if (!constructor.autoGas) return accept();
 
       interfaceAdapter
         .estimateGas(params)
         .then(gas => {
-          const bestEstimate = utils.multiplyBigNumberByDecimal(
-            utils.bigNumberify(gas),
-            constructor.gasMultiplier
-          );
-
-          // Don't go over blockLimit
+          // there are situations where the web3 gas estimation function in interfaceAdapter
+          // fails, specifically when a transaction will revert; we still want to continue
+          // the user flow for debugging purposes; so we provide a default gas for
+          // that situation, equal to half of the blockLimit for the latest block
           const limit = utils.bigNumberify(blockLimit);
+          let bestEstimate;
+          if(gas === null) {
```

Comment:
stupid nit but you are missing a space after `if`

Reply:
Fixed! :)